docs/DataTypes.md

```diff
@@ -0,0 +1,36 @@
+# Data Types
+
+DataHelix currently recognises three distinct data types. Keeping this set small is a deliberate goal; it would be possible to have types like _FirstName_ or _Postcode_, but instead these are considered specialisations of the _String_ type, so they can be constrained by the normal string operators (e.g. a user could generate all first names shorter than 10 characters, starting with a vowel).
```

Comment:
It could be beneficial to call out the absence of booelan support, but that it is on the roadmap to support them - #372

Reply:
I'm wary of predictive documentation - I think it's good when your planned course directly contradicts existing behaviour (as is the case with default granularities), but otherwise it feels kind of un-agile - we should assume that any planned work could be deprioritised or cancelled, and only document it when we actually do it.